browser/ui/ai_chat/tab_informer.h

```diff
@@ -0,0 +1,66 @@
+// Copyright (c) 2025 The Brave Authors. All rights reserved.
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this file,
+// You can obtain one at https://mozilla.org/MPL/2.0/.
+
+#ifndef BRAVE_BROWSER_UI_AI_CHAT_TAB_INFORMER_H_
+#define BRAVE_BROWSER_UI_AI_CHAT_TAB_INFORMER_H_
+
+#include <string_view>
+#include <vector>
+
+#include "base/memory/raw_ptr.h"
+#include "brave/components/ai_chat/core/common/mojom/tab_informer.mojom-forward.h"
+#include "brave/components/ai_chat/core/common/mojom/tab_informer.mojom.h"
+#include "chrome/browser/ui/browser_tab_strip_tracker.h"
+#include "chrome/browser/ui/browser_tab_strip_tracker_delegate.h"
+#include "chrome/browser/ui/tabs/tab_strip_model_observer.h"
+#include "content/public/browser/web_contents.h"
+#include "mojo/public/cpp/bindings/pending_receiver.h"
+#include "mojo/public/cpp/bindings/receiver.h"
+#include "mojo/public/cpp/bindings/remote_set.h"
+
+namespace ai_chat {
+
+class TabInformer : public mojom::TabInformer,
```

Comment:
Perhaps a unit test for this class

Reply:
Unfortunately this won't work in a unittest - it needs to be in a browsertest. I've added one :smile: